src/sentry/api/helpers/group_index/update.py

```diff
@@ -713,6 +705,28 @@ def update_groups(
     return Response(result)
 
 
+def get_latest_release(project: Project) -> Release | None:
+    release = None
+    follows_semver = follows_semver_versioning_scheme(
+        org_id=project.organization_id, project_id=project.id
+    )
+    releases = Release.objects.filter(projects=project, organization_id=project.organization_id)
+    if follows_semver:
+        release = (
+            releases.filter_to_semver()
+            .annotate_prerelease_column()
+            .order_by(*[f"-{col}" for col in Release.SEMVER_COLS])
```

Comment:
This logic is based on this code:
https://github.com/getsentry/sentry/blob/914c74114efdfcca7c3d5b4f29ff9308cc643bce/src/sentry/api/helpers/group_index/update.py#L141-L151